lxd/events.go

```diff
@@ -35,7 +35,31 @@ func (r *eventsServe) String() string {
 }
 
 func eventsSocket(d *Daemon, r *http.Request, w http.ResponseWriter) error {
+	allProjects := queryParam(r, "all-projects") == "true"
+	projectQueryParam := queryParam(r, "project")
+	if allProjects && projectQueryParam != "" {
+		response.BadRequest(fmt.Errorf("'project' cannot be set if 'all-projects' is set"))
+		return nil
+	}
+
+	if !allProjects && projectQueryParam != "" {
+		_, err := d.cluster.GetProject(projectQueryParam)
+		switch err {
```

Comment:
Please can you refactor this error checking to be like:

```go
if err != nil {
    if errors.Is(err, db.ErrNoSuchObject) {
       response.BadRequest(fmt.Errorf("Project %q not found", projectQueryParam)).Render(w)
    }

    return err
}
```

This has the following benefits:

* Its fewer lines to read.
* It will handle changes to `GetProject` in the future in case the error is wrapped.
* It doesn't need to handle the "happy-path" (no error) explicitly so its easier to read.